Mathlib/Analysis/SpecialFunctions/Integrals.lean

```diff
@@ -140,10 +140,7 @@ theorem intervalIntegrable_cpow {r : ℂ} (h : 0 ≤ r.re ∨ (0 : ℝ) ∉ [[a,
   · -- case `c < 0`: integrand is identically constant, *except* at `x = 0` if `r ≠ 0`.
     apply IntervalIntegrable.symm
     rw [intervalIntegrable_iff_integrableOn_Ioc_of_le hc.le]
-    have : Ioc c 0 = Ioo c 0 ∪ {(0 : ℝ)} := by
-      rw [← Ioo_union_Icc_eq_Ioc hc (le_refl 0), ← Icc_def]
-      simp_rw [← le_antisymm_iff, setOf_eq_eq_singleton']
-    rw [this, integrableOn_union, and_comm]; constructor
+    rw [← Ioo_union_right hc, integrableOn_union, and_comm]; constructor
```

Comment:
Here, `hc` has type `c < 0`, and [`Ioo_union_right`](https://github.com/leanprover-community/mathlib4/blob/16956ef06e49e3e4f34d7be6d97f170580b42a79/Mathlib/Order/Interval/Set/Basic.lean#L597) has type `Set.Ioo c 0 ∪ {0} = Set.Ioc c 0`.

(Just noting this to show that the change I'm proposing does not introduce any defeq abuse.)